lighthouse-core/test/report/html/renderer/util-test.js

```diff
@@ -142,4 +146,23 @@ describe('util helpers', () => {
     Util.formatDisplayValue(displayValue);
     assert.deepStrictEqual(displayValue, cloned, 'displayValue was mutated');
   });
+
+  describe('getFinalScreenshot', () => {
+    sampleResults.reportCategories = Object.values(sampleResults.categories);
+    const category = sampleResults.reportCategories.find(cat => cat.id === 'performance');
+    ReportRenderer.smooshAuditResultsIntoCategories(sampleResults.audits,
+      sampleResults.reportCategories);
+
+    it('gets a datauri as a string', () => {
+      const datauri = Util.getFinalScreenshot(category);
+      assert.equal(typeof datauri, 'string');
+      assert.ok(datauri.startsWith('data:image/jpeg;base64,'));
+    });
+
+    it('returns null if there are no screenshots', () => {
+      const fakeCategory = Object.assign({}, category, {auditRefs: []});
```

Comment:
can you pass in the `pwa` category or cancel out just the final screenshot audit (or both?)? It's hard to tell which error case this is hitting since the category shape is so smooshed up by this point :)